services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -15,14 +16,31 @@ export const ChoiceListField = ({
   ...props
 }: Props) => {
   const mqClasses = makeMediaQueryClasses();
+  const defaultChoices = choices.map((choice) => {
+    return { value: choice.value, checked: choice.defaultChecked || false };
+  });
+
+  const [choicesChosen, setChoicesChosen] =
+    useState<ChoiceListSelected[]>(defaultChoices);
+
+  useEffect(() => {
+    form.setValue(name, choicesChosen, { shouldValidate: true });
+  }, [choicesChosen]);
 
   // get the form context
   const form = useFormContext();
 
   // update form data
-  const onChangeHandler = async (event: InputChangeEvent) => {
-    const { name: choiceListName, value: choiceListValue } = event.target;
-    form.setValue(choiceListName, choiceListValue, { shouldValidate: true });
+  const onChangeHandler = (event: InputChangeEvent) => {
+    const choiceSelected: ChoiceListSelected = {
+      value: event.target.value,
+      checked: event.target.checked,
+    };
+    setChoicesChosen((prevState) => {
+      return prevState.map((choice) => {
+        return choice.value === choiceSelected.value ? choiceSelected : choice;
+      });
+    });
```

Comment:
The plan has been to make a `CheckboxField` component and a `RadioField` component, both comprised of a `ChoiceListField` component after the `ChoiceList` was done. We may want to do that in this PR and consider having separate `onChangeHandler` functions for each.

Reply:
Is there a different output we would like to have? Right now both Radio and Checkbox's return this format:

```
abf-radio: Array(3)
0: {value: 'A', checked: false}
1: {value: 'B', checked: true}
2: {value: 'C', checked: false}
```

Reply:
Settled on this format over a conversation with Brax and Garrett:

Radio:
```
abf-a1o1: Array(1)
0: {value: 'C', id: 'abf-radio03'}
```

Checkbox
```
abf-a2o1: Array(3)
0: {value: 'A', id: 'abf-checkboxo1'}
1: {value: 'B', id: 'abf-checkbox02'}
2: {value: 'C', id: 'abf-checkbox03'}
```

Both of these cases are only showing array values that are checked. Unchecked is not shown in the array.